src/lib/y2firstboot/clients/wsl.rb

```diff
@@ -60,6 +62,18 @@ def setup_machine_id
         # missing
         Yast::Execute.locally("/usr/bin/systemd-machine-id-setup")
       end
+
+      def switch_product
+        yaml_product = ::Registration::YamlProduct.selected_product
+        return unless yaml_product
+
+        return if yaml_product["name"] == "SLES" # sles is already selected in WSL
```

Comment:
As already discussed on IRC, use the installed product here.e.g.


```suggestion
        installed_product = Y2Packager::Resolvable.find(kind: :product, status: :installed, category: "base").first
        return if installed_product && yaml_product["name"] ==  installed_product.name
```